### PR TITLE
Added dbt parser

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix https://github.com/open-metadata/OpenMetadata/issues/9976

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 1]
```diff
@@ -629,154 +637,171 @@ def create_dbt_tests_suite_definition(
...
-            test_case_result = TestCaseResult(
-                timestamp=dbt_timestamp,
-                testCaseStatus=test_case_status,
-                testResultValue=[
-                    TestResultValue(
-                        name=dbt_test_result.get("unique_id"),
-                        value=str(test_result_value),
+                # Create the test case fqns and add the results
+                for table_fqn in dbt_test.get("upstream"):
+                    source_elements = table_fqn.split(".")
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:755]: We have a constant named `FQN_SEPARATOR` for this. Let's import it from https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/fqn.py#L48

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 2]
```diff
@@ -787,23 +812,17 @@ def generate_entity_link(self, dbt_test):
...
     def get_dbt_compiled_query(self, mnode) -> Optional[str]:
-        dbt_query_key_names = ["compiled_sql", "compiled_code"]
-        for key_name in dbt_query_key_names:
-            query = mnode.get(key_name)
-            if query:
-                return query
-        logger.debug(
-            f"Unable to get DBT compiled query for node - {mnode.get('name','unknown')}"
-        )
+        if hasattr(mnode, "compiled_code") and mnode.compiled_code:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:815]: same, these values that have specific and important meaning such as `compiled_code` and `compiled_sql` could be part of an Enum for `COMPILED_QUERIES` or smt.

**Review by pmbrull — CHANGES_REQUESTED** [review]
Let's clean a bit the magic strings into specific Enums.

Also, can we create constants for lists such as `required_manifest_keys = [
            "alias",
            "name",
            "schema",
            "database",
            "resource_type",
            "description",
        ]`?

We can add a comment linking to the dbt resources where they specify which values are required.

Finally, let's add a few unit tests where pieces of logic are not easy to follow, such as `validate_dbt_files`, `parse_upstream_nodes` and `yield_data_models` -> there might be code blocks that can be converted into smaller functions fo easier testing.

We should feel confident that we are not missing anything here.

Thanks for the effort

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 3]
```diff
@@ -433,28 +436,26 @@ def parse_data_model_columns(
...
-                column_name = manifest_column.get("name")
-                column_type = manifest_column.get("data_type")
-                column_description = manifest_columns.get(key.lower(), {}).get(
-                    "description"
-                )
-                dbt_column_tags = manifest_columns.get(key.lower(), {}).get("tags")
+                column_name = manifest_column.name
+                column_type = manifest_column.data_type
+                column_description = manifest_column.description
+                dbt_column_tags = manifest_column.tags
```
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:446]: Makes sense will improve the logic here

**Review by pmbrull — APPROVED** [review]

**Pull request merged**